app/src/main/java/chat/rocket/android/contacts/ContactsFragment.kt

```diff
@@ -241,49 +284,68 @@ class ContactsFragment : Fragment() {
             with((activity as AppCompatActivity?)?.supportActionBar) {
                 profileButton = this?.getCustomView()?.findViewById(R.id.profile_image_avatar)
                 profileButton?.visibility = View.GONE
+                onlineStatusButton=this?.getCustomView()?.findViewById(R.id.text_online)
+                onlineStatusButton?.visibility=View.GONE
                 widechatSearchView = this?.getCustomView()?.findViewById(R.id.action_widechat_search)
                 widechatSearchView?.visibility = View.GONE
             }
         }
     }
 
-    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
-        super.setUserVisibleHint(isVisibleToUser)
-    }
 
     fun setupFrameLayout(filteredContactArrayList: ArrayList<Contact>) {
-        try {
-            val contactListFragment = ContactListFragment.newInstance(
-                    filteredContactArrayList,
-                    contactHashMap
-            )
-            val fragmentTransaction = childFragmentManager.beginTransaction()
-            fragmentTransaction.replace(
-                    R.id.contacts_area,
-                    contactListFragment,
-                    "CONTACT_LIST_FRAGMENT"
-            )
-            fragmentTransaction.commit()
-        } catch (exception: IllegalStateException) {
-            //This is one bad user who clicks too fast
-        } catch (exception: NullPointerException) {
+
+
+        if (filteredContactArrayList!!.size == 0) {
+            emptyTextView!!.visibility = View.VISIBLE
+            recyclerView!!.visibility = View.GONE
+        } else {
+            emptyTextView!!.visibility = View.GONE
+            recyclerView!!.visibility = View.VISIBLE
+
+            recyclerView!!.setHasFixedSize(true)
+            recyclerView!!.layoutManager = LinearLayoutManager(context)
+            recyclerView!!.adapter = ContactRecyclerViewAdapter(this.activity as MainActivity, map(filteredContactArrayList)!!, contactHashMap)
         }
+    }
+
+
 
 
+    fun map(contacts: List<Contact>): ArrayList<ItemHolder<*>> {
```

Comment:
Here is my suggestion for this function:

fun map(contacts: List<Contact>): ArrayList<ItemHolder<*>> {
        val finalList = ArrayList<ItemHolder<*>>(contacts.size + 2)
        val userList = ArrayList<ItemHolder<*>>(contacts.size)
        val contactsList = ArrayList<ItemHolder<*>>(contacts.size)

        contacts.forEach { contact ->
            if(contact.getUsername()!=null){
                userList.add(ContactItemHolder(contact))
            }else{
                contactsList.add(ContactItemHolder(contact))
            }
        }
        finalList.addAll(userList)
        finalList.add(ContactHeaderItemHolder("INVITE CONTACTS"))
        finalList.addAll(contactsList)
        finalList.add(inviteItemHolder("invite"))

        return finalList
    }

Reply:
done